### ISSUE TITLE
Add support for Python 3.10 in CI

### ISSUE THREAD

**Issue by aaugustin** [aaugustin/websockets#935]
When Python 3.10 is released.

This requires updating:

* .appveyor.yml
* .circleci/config.yml
* .travis.yml
* tox.ini

This was missed when adding 3.9.

**Comment by aaugustin** [comment]
Blocked until a release of Python 3.10 is available in https://github.com/actions/python-versions/releases

(I do not wish to keep track of alphas and betas.)

**Comment by GanerCodes** [comment]
3.10 is essentially finalized besides some possible bug fixes/security exploits if they are discovered, so starting has no risk of python breaking your code.

**Comment by aaugustin** [comment]
This issue is only about configuring the GitHub Actions CI. I cannot do this before GitHub Actions supports 3.10.

websockets already works with Python 3.10 in the master branch.

**Comment by kumaraditya303** [comment]
@aaugustin Python 3.10rc1 is out and Github Actions supports it now https://github.com/actions/python-versions/blob/41e5bb2c77999811a99fa634bafe9537fee86548/versions-manifest.json#L14

**Comment by aaugustin** [comment]
Apparently not, see #1030.

I assume I could put something other than "3.10" in the config and it would pick 3.10rc1 instead of 3.1. However, as explained above, I don't wish to update the CI config for every prerelease of Python, so I'll wait until the real 3.10 is out.

**Comment by kumaraditya303** [comment]
> Apparently not, see #1030.
> 
> I assume I could put something other than "3.10" in the config and it would pick 3.10rc1 instead of 3.1. However, as explained above, I don't wish to update the CI config for every prerelease of Python, so I'll wait until the real 3.10 is out.

I have suggested the change in the PR to fix it.

**Comment by aaugustin** [comment]
For the third time: I will add "3.10" when that works.

**Comment by kumaraditya303** [comment]
If you use `3.10-dev` it will automatically pick the latest python 3.10 prerelease as I suggested in the PR and when the final python `3.10` is released you can change it to `3.10`. IMO with this you have to update it once when the final is released and prelease will works automatically.

**Comment by aaugustin** [comment]
Yes, that's true. However, my experience of the last decade proves that I will forget to update it when the final is released. I decided several years ago to keep Python version matrices simple; that decision serves me well and never caused trouble; I'm sticking with it.

**Comment by kumaraditya303** [comment]
> Yes, that's true. However, my experience of the last decade proves that I will forget to update it when the final is released. I decided several years ago to keep Python version matrices simple; that decision serves me well and never caused trouble; I'm sticking with it.

In that case you use a version range which does not need to be changed even after the latest version is release like this 
`3.10.0-alpha - 3.10.0`

**Comment by aaugustin** [comment]
I appreciate you're trying to help, but decades of programming experience taught me the value of simplicity, and the simple solution here is just to put "3.10" when GitHub supports it.

I'm working with a 3.10 prerelease locally so I'll notice any issue quickly; CI checks other versions; really there's no problem that need solving.
